modules/core/02-client/abci.go

```diff
@@ -1,42 +0,0 @@
-package client
-
-import (
-	sdk "github.com/cosmos/cosmos-sdk/types"
-
-	"github.com/cosmos/ibc-go/v3/modules/core/02-client/keeper"
-	"github.com/cosmos/ibc-go/v3/modules/core/exported"
-	ibctmtypes "github.com/cosmos/ibc-go/v3/modules/light-clients/07-tendermint/types"
-)
-
-// BeginBlocker updates an existing localhost client with the latest block height.
-func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
-	plan, found := k.GetUpgradePlan(ctx)
-	if found {
-		// Once we are at the last block this chain will commit, set the upgraded consensus state
-		// so that IBC clients can use the last NextValidatorsHash as a trusted kernel for verifying
-		// headers on the next version of the chain.
-		// Set the time to the last block time of the current chain.
-		// In order for a client to upgrade successfully, the first block of the new chain must be committed
-		// within the trusting period of the last block time on this chain.
-		_, exists := k.GetUpgradedClient(ctx, plan.Height)
-		if exists && ctx.BlockHeight() == plan.Height-1 {
-			upgradedConsState := &ibctmtypes.ConsensusState{
-				Timestamp:          ctx.BlockTime(),
-				NextValidatorsHash: ctx.BlockHeader().NextValidatorsHash,
-			}
-			bz := k.MustMarshalConsensusState(upgradedConsState)
-
-			k.SetUpgradedConsensusState(ctx, plan.Height, bz)
-		}
-	}
-
```

Comment:
this is not related to the localhost

Reply:
https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/abci.go#L11

This looks to me like it's only being used by the localhost client? (comments also suggest as much)

Reply:
https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/abci.go#L11

This looks to me like it's only being used by the localhost client? (comments also suggest as much)

Reply:
No the godoc is incorrect. This code is required to perform IBC client upgrades, see in code documentation

Reply:
Ah right, got it thanks. I'll update the godoc :+1: